tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/effects/debugger_effects.ts

```diff
@@ -638,8 +641,9 @@ export class DebuggerEffects {
             // TODO(cais): Do this reshaping in the backend and simplify
             // the frontend code here.
             for (let i = 0; i < stackFrameIds.length; ++i) {
-              stackFramesById[stackFrameIds[i]] =
-                stackFramesResponse.stack_frames[i];
+              stackFramesById[stackFrameIds[i]] = stackFrameAsArray2StackFrame(
+                stackFramesResponse.stack_frames[i]
+              );
```

Comment:
Convert the backend data structure into the FE at the data source level, please.